### PR TITLE
Internal: Added more template tests. Updated old one. [ED-10352]

### PR DESCRIPTION
**Pull request by dmaziar**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x ] Other... Please describe: more templates tests for Core.

## Summary
Added  'icon', 'image_box', 'image_carousel' tests. Updated existing one to align with testim coverage.

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: tests/elements-regression/tests/elements-regression.test.js [hunk 1]
```diff
@@ -11,18 +11,24 @@ const image2 = {
 	extension: 'jpg',
 };
 
+const image1 = {
+	title: 'image1',
+	extension: 'jpg',
+};
+
```
Comment (yotamselementor) [inline tests/elements-regression/tests/elements-regression.test.js:18]: Place this before image2 for better readability

File: tests/playwright/pages/editor-page.js [hunk 2]
```diff
@@ -17,8 +17,30 @@ module.exports = class EditorPage extends BasePage {
...
+			if ( 'carousel' in el.settings ) {
+				for ( let i = 0; i < el.settings.carousel.length; i++ ) {
+					const newUrl = el.settings.carousel[ i ].url.replace( regex, `${ date.getFullYear() }/${ month }` );
+					el.settings.carousel[ i ].url = newUrl;
+				}
+			}
+		} );
+	}
+
+	async loadTemplate( filePath, updateDatesForImages = false ) {
```
Comment (yotamselementor) [inline tests/playwright/pages/editor-page.js:39]: When do we use `updateDatesForImages=true`? Maybe add some short documentation for this function?

File: tests/playwright/pages/editor-page.js [hunk 3]
```diff
@@ -17,8 +17,30 @@ module.exports = class EditorPage extends BasePage {
...
+				el.settings.image.url = newUrl;
+			}
+			if ( 'carousel' in el.settings ) {
+				for ( let i = 0; i < el.settings.carousel.length; i++ ) {
+					const newUrl = el.settings.carousel[ i ].url.replace( regex, `${ date.getFullYear() }/${ month }` );
+					el.settings.carousel[ i ].url = newUrl;
+				}
+			}
+		} );
+	}
```
Comment (yotamselementor) [inline tests/playwright/pages/editor-page.js:37]:
```suggestion
	updateImageDates( templateData ) {
		const replaceUrl = ( url ) => {
			const date = new Date();
			const month = date.toLocaleString( 'default', { month: '2-digit' } );
			const regex = /[0-9]{4}\/[0-9]{2}/g;
			return url.replace( regex, `${ date.getFullYear() }/${ month }` );
		};

		templateData.content[ 0 ].elements.forEach( ( el ) => {
			if ( 'image' in el.settings ) {
				el.settings.image.url = replaceUrl( el.settings.image.url );
			}
			if ( 'carousel' in el.settings ) {
				for ( let i = 0; i < el.settings.carousel.length; i++ ) {
					el.settings.carousel[ i ].url = replaceUrl( el.settings.carousel[ i ].url );
				}
			}
		} );
	}
```

**Review by yotamselementor — CHANGES_REQUESTED** [review]

File: tests/elements-regression/tests/elements-regression.test.js (hunk 1 above)
Comment (dmaziar) [reply tests/elements-regression/tests/elements-regression.test.js:14]: fixed

File: tests/playwright/pages/editor-page.js (hunk 2 above)
Comment (dmaziar) [reply tests/playwright/pages/editor-page.js:39]: added some description

**Review by yotamselementor — APPROVED** [review]

**Pull request merged**